### PR TITLE
parser: support ANSI_QUOTES sql mode

### PR DESCRIPTION
**Pull request by tiancaiamao**
In ANSI_QUOTES sql mode, "xxx" is parsed to  identifier,
similar to `xxx` in normal sql mode.

### CODE REVIEW

File: parser/lexer.go [hunk 1]
```diff
@@ -104,6 +108,11 @@ func (s *Scanner) Lex(v *yySymType) int {
 			tok = tok1
 		}
 	}
+	if (s.sqlMode&mysql.ModeANSIQuotes) > 0 &&
```
Comment (shenli) [inline parser/lexer.go]: Should we move this before line 103?

Comment (tiancaiamao) [inline parser/lexer.go]: I think identifier within "" would always be a kind of identifier token.
This is the difference: select will be rewrite to kind of token named select, while `select` is always remain a identifier token.
In ANSI_QUOTES mode, "select" behavior is similar to `select`, so no rewrite is needed (line 104).

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**